main/inc/lib/sessionmanager.lib.php

```diff
@@ -5543,4 +5538,405 @@ public static function getTotalUserCoursesInSession($sessionId)
+     * where course id_course is in sessions id_session1, id_session2
+     * @param $userId
+     * @return array
+     */
+    public static function getCoursesForCourseSessionCoach($userId)
+    {
+        $listResCourseSession = array();
+        $tblCourse = Database::get_main_table(TABLE_MAIN_COURSE);
+        $tblSessionRelCourseRelUser = Database::get_main_table(TABLE_MAIN_SESSION_COURSE_USER);
+        $tblSession = Database::get_main_table(TABLE_MAIN_SESSION);
+
+        $sql = "SELECT id_session, course_code, c.id
+                FROM $tblSessionRelCourseRelUser srcru
+
+                LEFT JOIN $tblCourse c
+                ON c.code = srcru.course_code
+
+                WHERE
+                srcru.id_user=".Database::escape_string($userId)."
+                AND srcru.status = 2";
+
+        $res = Database::query($sql);
+
+        while ($data = Database::fetch_assoc($res)) {
+            if (self::isSessionDateOkForCoach($data['id_session'])) {
+                if (!isset($listResCourseSession[$data['id']])) {
+                    $listResCourseSession[$data['id']] = array();
+                }
+                $listResCourseSession[$data['id']][] = $data['id_session'];
+            }
+        }
+        return $listResCourseSession;
+    }
+
+
+    /**
+     * Return true if coach is allowed to access this session
+     * @param $inSessionId
+     * @param $inCoachId
+     * @return bool
+     */
+    public static function isSessionDateOkForCoach($inSessionId)
+    {
+        $listSessionInfo = self::getSessionInfoById($inSessionId);
+        $dateStart = $listSessionInfo['date_start'];
+        $dateEnd = $listSessionInfo['date_end'];
+        $nbDaysAccessBeforeBeginning = $listSessionInfo['nb_days_access_before_beginning'];
+        $nbDaysAccessAfterEnd = $listSessionInfo['nb_days_access_after_end'];
+
+        // no start date
+        if ($dateStart == '0000-00-00') {
+            return true;
+        }
+
+        $now = time();
+
+        $dateStartForCoach = api_strtotime($dateStart.' 00:00:00') - ($nbDaysAccessBeforeBeginning * 86400);
+        $dateEndForCoach = api_strtotime($dateEnd.' 00:00:00') + ($nbDaysAccessAfterEnd * 86400);
+
+        if ($dateEnd == '0000-00-00') {
+            // start date but no end date
+            if ($dateStartForCoach <= $now) {
+                return true;
+            }
+        } else {
+            // start date and end date
+            if ($dateStartForCoach <= $now && $now <= $dateEndForCoach) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+
+    /**
+     * Return database info for session with id $inSessionId
+     * @param $sessionId
+     * @return array
+     */
+    public function getSessionInfoById($sessionId)
+    {
+        $tblSession = Database::get_main_table(TABLE_MAIN_SESSION);
+        $sql = "SELECT * FROM $tblSession WHERE id=".Database::escape_string($sessionId);
```

Comment:
Although this function will be removed, when using an integer value, you cannot escape it with "Database::escape_string()", this would be unsafe when using quotes. Please use intval() instead (only for integer values)